pkg/reconciler/reconciler.go

```diff
@@ -100,13 +103,21 @@ func NewBase(ctx context.Context, controllerAgentName string, cmw configmap.Watc
 		}()
 	}
 
+	// Create metrics reporter
+	logger.Debug("Creating stats reporter")
+	statsReporter, err := NewStatsReporter(controllerAgentName)
+	if err != nil {
+		logger.Fatal(err)
```

Comment:
Does this exit the controller? Is that the desired behavior or should we log an error and then proceed without metrics?